modules/drivers/sqlserver/test/metabase/driver/sqlserver_test.clj

```diff
@@ -644,6 +644,56 @@
                   (is (= expected-types
                          (map :base_type results-metadata-cols))))))))))))
 
+(deftest ^:parallel comparison-expression-in-custom-column-test
+  (mt/test-driver :sqlserver
+    (testing "Comparison expressions in custom columns are wrapped in CASE statements (#53805)"
+      (testing "Greater than comparison between two fields"
+        (let [query (mt/mbql-query orders
+                      {:expressions {"subtotal-gt-total" [:> $subtotal $total]}
+                       :fields      [[:expression "subtotal-gt-total"]]
+                       :limit       1})]
+          (testing "Should generate CASE WHEN ... THEN 1 ELSE 0 END"
+            (is (= ["SELECT"
+                    "  TOP(1) CASE"
+                    "    WHEN dbo.orders.subtotal > dbo.orders.total THEN 1"
+                    "    ELSE 0"
+                    "  END AS subtotal - gt - total"
+                    "FROM"
+                    "  dbo.orders"]
```

Comment:
This is overfitting; better to grep the SQL for just the `CASE` part and ignore newlines.